python/requirements.txt

```diff
@@ -1,3 +1,48 @@
-astropy
-fastapi
-mangum
+annotated-types==0.6.0
+antares-client==1.5.0
+anyio==3.7.1
+architect-functions==1.0.0
+astropy==6.0.0
+astropy-iers-data
+boto3==1.34.2
+botocore==1.34.2
+bson==0.5.10
+certifi==2023.11.17
+cffi==1.16.0
+charset-normalizer==3.3.2
+click==8.1.7
+confluent-kafka==1.7.0
```

Comment:
I don't see any use of confluent-kafka here. Why is it listed as a dependency?

Reply:
It's a dependency of antares-client.

Reply:
...and I see the same issue I had is causing the python github workflow to fail.

Reply:
Right, so the CI/CD pipeline is failing because there aren't wheels on PyPI for that version of confluent-kafka-python. I suggest opening a PR with antares_client to unpin their dependency so that downstream projects (like us) can use more recent versions.

But what I'm trying to get at, is how are you generating this list of dependencies? Is this a lock file representing everything in your Python environment, captured with `pip freeze`?

Reply:
I'm creating a new Python virtualenv, installing only the modules needed for the API to work, and then dumping them to `requirements.txt` with `pip freeze`, yes. 

I submitted an issue to the ANTARES client GitLab page about the version of confluent-kafka they're using.

Reply:
> I'm creating a new Python virtualenv, installing only the modules needed for the API to work, and then dumping them to `requirements.txt` with `pip freeze`, yes.

Alright. So up until now I have been lazy about this because we had very few dependencies, but I would now like to pick a Python dependency management tool that provides functionality comparable to NPM's package.json (specifying constraints on compatible versions of dependencies) and package-lock.json (specifying concrete resolved versions of dependencies). So we need to do a miniature trade study for that and pick one, probably in a new issue and PR.

> I submitted an issue to the ANTARES client GitLab page about the version of confluent-kafka they're using.

Thank you. Would you please add a link to it here?

Reply:
https://gitlab.com/nsf-noirlab/csdc/antares/client/-/issues/8

Reply:
https://gitlab.com/nsf-noirlab/csdc/antares/client/-/merge_requests/32

Reply:
@lpsinger this comment probably sheds some light as to why they pin the dependency: https://gitlab.com/nsf-noirlab/csdc/antares/client/-/merge_requests/22#note_1532581942

Reply:
Pretty sure that is fixed upstream by now, as the most recent versions of confluent-kafka-python definitely do validate TLS certificates.

What problem is this currently causing right now? The Lambda doesn't have a dependency on gcn-kafka-python at the moment, does it?

Reply:
Right now the problem is that antares-client won't build in the CI/CD workflow, because of the dependency on an old version of confluent-kafka module. This is because it compiles from source and needs librdkafka installed to build correctly.

Reply:
Ahhh, I see.

Reply:
OK, in that case, let's use your original code for Antares. But would you please add a FIXME comment explaining the problem and including a link to the upstream issue?